pkg/sources/git/git.go

```diff
@@ -964,30 +970,66 @@ func getSafeRemoteURL(repo *git.Repository, preferred string) string {
 	return safeURL
 }
 
-func handleBinary(ctx context.Context, repo *git.Repository, reporter sources.ChunkReporter, chunkSkel *sources.Chunk, commitHash plumbing.Hash, path string) error {
+func handleBinary(ctx context.Context, gitDir string, reporter sources.ChunkReporter, chunkSkel *sources.Chunk, commitHash plumbing.Hash, path string) error {
 	ctx.Logger().V(5).Info("handling binary file", "path", path)
-	commit, err := repo.CommitObject(commitHash)
-	if err != nil {
-		return err
+
+	if common.SkipFile(path) {
+		ctx.Logger().V(5).Info("skipping binary file", "path", path)
+		return nil
 	}
 
-	file, err := commit.File(path)
+	const (
+		catFileArg = "cat-file"
+		blobArg    = "blob"
+
+		maxSize = 100 * 1024 * 1024 // 100 MB
+	)
+	cmd := exec.Command("git", "--git-dir="+gitDir, catFileArg, blobArg, commitHash.String()+":"+path)
```

Comment:
Are `--git-dir` and `-C` doing the same thing?

Reply:
That's a great question, I think so. I can use -C instead.

Reply:
TIL: https://stackoverflow.com/a/42299983